app/models/institutions_archive.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require_dependency 'institution'
+
+class InstitutionsArchive < Institution
+  self.table_name = 'institutions_archives'
+
+  # class methods
+  def self.archive(version)
+    version_number = version.number
+
+    conn = ActiveRecord::Base.connection
+
+    begin
+      ActiveRecord::Base.transaction do
+        create_archives(version_number, conn)
+        cleanup_institutions(version_number, conn)
+      end
+    rescue ActiveRecord::StatementInvalid => e
+      notice = 'There was an error occurring at the database level'
+      error_msg = e.original_exception.result.error_message
+      Rails.logger.error "#{notice}: #{error_msg}"
+    rescue StandardError => e
+      notice = 'There was an error of unexpected origin'
+      error_msg = e.message
+      Rails.logger.error "#{notice}: #{error_msg}"
+    end
+
+    conn.close
+  end
+
+  def self.create_archives(version_number, conn)
+    str = <<-SQL
+      INSERT INTO institutions_archives
+        SELECT *
+          FROM institutions
+          WHERE version < ? ;
```

Comment:
I think we should make this `WHERE version = ?` to reduce the size of this operation.  Then we can control the execution of these statements through another mechanism (e.g. manually, or a loop)

Reply:
This would delete the current production version (i.e. the one being published) as `version_number` comes from https://github.com/department-of-veterans-affairs/gibct-data-service/blob/71a099c64eba46ecaea6682c51ae871f9364d0fc/app/controllers/dashboards_controller.rb#L45

Reply:
Changing the https://github.com/department-of-veterans-affairs/gibct-data-service/blob/ed4e0bde150bd8fe8bbb94a9c7deec71c80c652f/app/models/institutions_archive.rb#L9 to no longer have a parameter and just do below for clarity

```
    production_version = Versions.current_production.number
```

Reply:
Changed archiving process to now archive all versions that are less than newest production version and greater than or equal to previous production version

```
delete from institution 
    where version >= <previous_production_version> and version < <current_production_version>
```